### DISCUSSION TITLE
`add_route` types missing one use case

### DISCUSSION THREAD

**Discussion by patrick91** [encode/starlette#2635] [Potential Issue] [answered]
Hi there! In Strawberry we have an ASGI view that can be used with Starlette, like this

```python
from starlette.applications import Starlette
from strawberry.asgi import GraphQL

from api.schema import schema

graphql_app = GraphQL(schema)

app = Starlette()
app.add_route("/graphql", graphql_app)
```

This route is a callable that accepts `scope`, `receive` and `send`, but looks like the types of `add_route` don't support that use case:

https://github.com/encode/starlette/blob/6f863b0d3b8e8f18d5df9e8cd2514f7085b874e1/starlette/applications.py#L161

is this expected? if not I'm happy to send a PR to update the types :)

**Comment by Kludex** [comment]
It is expected. Why is the app not mounted?

**Reply by patrick91** [reply]
@Kludex mmh, not sure I follow, do we need to change our implementation? 😊

**Reply by Kludex** [reply, marked as answer]
> @Kludex mmh, not sure I follow, do we need to change our implementation? 😊

I don't think so... But on the docs, you can just have the "mount" as follows:

```py
from starlette.applications import Starlette
from strawberry.asgi import GraphQL

from api import schema

graphql_app = GraphQL[None, None](schema)

app = Starlette()
app.mount("/graphql", graphql_app)
```

Or even better:

```py
from starlette.applications import Starlette
from starlette.routing import Mount
from strawberry.asgi import GraphQL

from api import schema

graphql_app = GraphQL[None, None](schema)

app = Starlette(routes=[Mount("/graphql", graphql_app)])
```
